cmd/gaia/app/genesis.go

```diff
@@ -167,43 +84,63 @@ func GaiaAppGenState(cdc *codec.Codec, appGenTxs []json.RawMessage) (genesisStat
 
 	// start with the default staking genesis state
 	stakeData := stake.DefaultGenesisState()
-
-	// get genesis flag account information
 	genaccs := make([]GenesisAccount, len(appGenTxs))
-	for i, appGenTx := range appGenTxs {
 
-		var genTx GaiaGenTx
-		err = cdc.UnmarshalJSON(appGenTx, &genTx)
+	for i, genTx := range appGenTxs {
+		var tx auth.StdTx
+		err = cdc.UnmarshalJSON(genTx, &tx)
 		if err != nil {
 			return
 		}
+		msgs := tx.GetMsgs()
+		if len(msgs) == 0 {
+			err = errors.New("must provide at least genesis message")
+			return
+		}
+		msg := msgs[0].(stake.MsgCreateValidator)
 
 		// create the genesis account, give'm few steaks and a buncha token with there name
-		genaccs[i] = genesisAccountFromGenTx(genTx)
-		stakeData.Pool.LooseTokens = stakeData.Pool.LooseTokens.Add(sdk.NewDecFromInt(freeFermionsAcc)) // increase the supply
+		genaccs[i] = genesisAccountFromMsgCreateValidator(msg)
+		stakeData.Pool.LooseTokens = stakeData.Pool.LooseTokens.Add(sdk.NewDecFromInt(msg.Delegation.Amount)) // increase the supply
 
 		// add the validator
-		if len(genTx.Name) > 0 {
-			stakeData = addValidatorToStakeData(genTx, stakeData)
-		}
+		//if len(msg.Description.Moniker) > 0 {
+		//	stakeData = addValidatorToStakeData(msg, stakeData)
+		//}
+
 	}
 
 	// create the final app state
 	genesisState = GenesisState{
 		Accounts:  genaccs,
+		Txs:       appGenTxs,
 		StakeData: stakeData,
 		GovData:   gov.DefaultGenesisState(),
 	}
 
 	return
 }
 
-func addValidatorToStakeData(genTx GaiaGenTx, stakeData stake.GenesisState) stake.GenesisState {
-	desc := stake.NewDescription(genTx.Name, "", "", "")
-	validator := stake.NewValidator(
-		sdk.ValAddress(genTx.Address), sdk.MustGetConsPubKeyBech32(genTx.PubKey), desc,
-	)
+func DefaultState(moniker string, pubKey crypto.PubKey) (genesisState GenesisState, genValidator tmtypes.GenesisValidator) {
+	acc := NewDefaultGenesisAccount(sdk.AccAddress(pubKey.Address()))
+	stakeData := stake.DefaultGenesisState()
+	validator := stake.NewValidator(sdk.ValAddress(acc.Address), pubKey, stake.NewDescription(moniker, "", "", ""))
+	stakeData = addValidatorToStakeData(validator, stakeData)
+	stakeData.Pool.LooseTokens = stakeData.Pool.LooseTokens.Add(sdk.NewDecFromInt(freeFermionsAcc))
```

Comment:
just reading the code, it looks like it's being done twice here?  once in the function `addValdiatorToStakeData` above.